src/Surfnet/ServiceProviderDashboard/Application/Service/EntityService.php

```diff
@@ -106,4 +107,29 @@ public function getEntityListForService(Service $service)
 
         return new ViewObject\EntityList($entities);
     }
+
+    /**
+     * @param string $manageId
+     * @param string $env
+     *
+     * @return array|null
+     *
+     * @throws InvalidArgumentException
+     * @throws \Surfnet\ServiceProviderDashboard\Infrastructure\Manage\Exception\QueryServiceProviderException
+     */
+    public function getManageEntityById($manageId, $env = 'test')
```

Comment:
In this specific case, I don't agree it would be actually better. Also for the implications of this action. It would require a new route for the delete action, or add a similar switch statement in the controller.